### PR TITLE
fix(core): action-sheet, fix a11y issues

### PR DESCRIPTION
**Pull request by Betrozov**
## Related Issue

closes https://github.com/sap/fundamental-ngx/issues/6211

## Description

Fixed Action Sheet i11y issues

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [n/a] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [n/a] check for console logs (warnings, errors)
-   [n/a] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [v] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/action-sheet/examples/action-sheet-mobile/action-sheet-mobile-example.component.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div style="position: relative">
     <fd-action-sheet [mobile]="true">
         <fd-action-sheet-control>
-            <button fd-button glyph="settings"></button>
+            <button fd-button glyph="settings" title="Open Action Sheet Mobile" aria-haspopup="menu"></button>
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/action-sheet/examples/action-sheet-mobile/action-sheet-mobile-example.component.html:4]: Great work! I was looking at this documentation: https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

and you need `aria-controls` for the button and an id for the menu it controls. You will need an `id` for the button as well, as you will need it for the `aria-labelledby` attr of the ul.

**Review by platon-rov — APPROVED** [review]

**Review by dmitry-stepanenko — APPROVED** [review]

File: libs/core/src/lib/action-sheet/action-sheet-item/action-sheet-item.component.ts [hunk 2]
```diff
@@ -36,7 +36,8 @@ export interface ActionSheetClickEvent {
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush,
     host: {
-        class: 'fd-action-sheet__item'
+        class: 'fd-action-sheet__item',
+        '[attr.role]': "'none'"
```
Comment (dimamarksman) [inline libs/core/src/lib/action-sheet/action-sheet-item/action-sheet-item.component.ts:40]:
```suggestion
        role: 'none'
```

File: libs/core/src/lib/action-sheet/action-sheet-item/action-sheet-item.component.ts [hunk 3]
```diff
@@ -52,11 +53,11 @@ export class ActionSheetItemComponent implements KeyboardSupportItemInterface {
     @Input()
     negative = false;
 
-    /** Indicate if items should be in compact or compare mode. **/
+    /** Indicate if items should be in compact or compare mode. */
```
Comment (dimamarksman) [inline libs/core/src/lib/action-sheet/action-sheet-item/action-sheet-item.component.ts:56]:
```suggestion
    /** Indicate if items should be in compact mode. */
```

**Review by dimamarksman — CHANGES_REQUESTED** [review]
Looks good.
I've noticed that closing a menu by Escape key leads to focus loss.

**Review by dimamarksman — APPROVED** [review]

**Comment by Veerakarthick** [comment]
Hello Colleague,

Retest has been taken below are the details.

Issue1: Fixed
Issue2: Fixed
Issue3: Fixed
Issue4: Fixed
Issue5: Fixed
Issue6:Fixed.

Retest Environment:
URL: https://deploy-preview-7390--fundamental-ngx.netlify.app/assets/typedoc/core/classes/actionsheetitemcomponent
Browser: Chrome Version 96.0.4664.110 (Official Build) (64-bit)
Screen reader: Jaws2021.2111.13ILM


Regards,
Karthikeyan.c

**Comment by Veerakarthick** [comment]
Hello Colleague,

Retest has been taken below are the details.

Issue1: Fixed
Issue2: Fixed
Issue3: Fixed
Issue4: Fixed
Issue5: Fixed
Issue6:Fixed.

Retest Environment:
URL: https://deploy-preview-7390--fundamental-ngx.netlify.app/assets/typedoc/core/classes/actionsheetitemcomponent
Browser: Chrome Version 96.0.4664.110 (Official Build) (64-bit)
Screen reader: Jaws2021.2111.13ILM


Regards,
Karthikeyan.c

**Pull request merged**